File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0 
        right = len(numbers) - 1
        res = []
        while left < right:
            s = numbers[left] + numbers[right]
            if s > target:
                right = right - 1 
            elif s < target:
                left = left + 1 
            else:
                res.append(left+1)
                res.append(right+1)
                break;
        return res

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0 ;
        int right = heights.size() - 1;
        int res = 0;
        int area ;
        while(left < right){            
            area = min(heights[left],heights[right]) * (right - left);
            if(heights[left] < heights[right]){
                left++ ;
                cout<<heights[left]<<" left ";
                cout<<heights[right-1]<<" right ";
            }
            else {
                right-- ;
            }
            
            res = max(area,res);
        
        }        
        return res;
    }
};
